tabular/src/autogluon/tabular/predictor/predictor.py

```diff
@@ -3132,14 +3140,24 @@ def _set_feature_generator(self, feature_generator='auto', feature_metadata=None
                                                                         init_kwargs=init_kwargs)
 
     def _sanitize_stack_args(self, num_bag_folds, num_bag_sets, num_stack_levels, time_limit, auto_stack,
-                             num_train_rows):
+                             num_train_rows, problem_type):
         if auto_stack:
             # TODO: What about datasets that are 100k+? At a certain point should we not bag?
             # TODO: What about time_limit? Metalearning can tell us expected runtime of each model, then we can select optimal folds + stack levels to fit time constraint
             if num_bag_folds is None:
                 num_bag_folds = min(8, max(5, math.floor(num_train_rows / 100)))
+            # TODO: Leverage use_bag_holdout when data is large to enable multi-layer stacking
+            #  if num_train_rows >= 100000 and num_val_rows is None and use_bag_holdout is None:
+            #      use_bag_holdout = True
             if num_stack_levels is None:
-                num_stack_levels = min(1, max(0, math.floor(num_train_rows / 750)))
+                if problem_type == BINARY:
+                    # Disable multi-layer stacking to avoid stack info leakage
+                    num_stack_levels = 0
```

Comment:
Just ask, should we use `num_stack_levels=0` or `num_stack_levels=1`?

Reply:
0 = one-layer stacking (L1 models, L2 Weighted), 1 = multi-layer stack with L1 and L2 models (and L3 Weighted), 2 = L1,L2,L3 models + L4 Weighted, etc.

I know that it is a bit confusing, but I think this works, since 0 == a weighted combination of base models with no complex stacker models.

I also want to avoid changing cause it will be a breaking change.

Reply:
Yeah, that makes sense.